src/Symfony/Bridge/Doctrine/ArgumentResolver/EntityValueResolver.php

```diff
@@ -176,59 +162,55 @@ private function getIdentifier(Request $request, array $options, string $name):
             return $id;
         }
 
-        if (null !== $options['id']) {
-            $name = $options['id'];
+        if (null !== $options->id) {
+            $name = $options->id;
         }
 
         if ($request->attributes->has($name)) {
             return $request->attributes->get($name);
         }
 
-        if (!$options['id'] && $request->attributes->has('id')) {
+        if (!$options->id && $request->attributes->has('id')) {
             return $request->attributes->get('id');
         }
 
         return false;
     }
 
-    private function findOneBy(string $class, Request $request, array $options): false|object|null
+    private function findOneBy(string $class, Request $request, MapEntity $options): false|object|null
     {
-        if (!$options['mapping']) {
-            if (!$options['auto_mapping']) {
```

Comment:
No need for option `auto_mapping`: making `mapping` nullable does the job.

Reply:
good catch